packages/fxa-settings/src/lib/channels/helpers.ts

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { AccountData } from '../../models';
+
+const ALLOWED_LOGIN_FIELDS = [
+  'declinedSyncEngines',
+  'email',
+  'keyFetchToken',
+  'offeredSyncEngines',
+  'sessionToken',
+  'services',
+  'uid',
+  'unwrapBKey',
+  'verified',
+];
+
+const REQUIRED_LOGIN_FIELDS = [
+  'email',
+  'keyFetchToken',
+  'sessionToken',
+  'uid',
+  'unwrapBKey',
+  'verified',
+];
+
+// Might need to go on the Integration?
+// let uidOfLoginNotification: hexstring = '';
+
+// function hasRequiredLoginFields(loginData) {
+//   const loginFields = Object.keys(loginData);
+//   return (
+//     REQUIRED_LOGIN_FIELDS.filter((field) => !loginFields.includes(field))
+//       .length === 0
+//   );
+// }
+
+/**
+ * Get login data from `account` to send to the browser.
+ * All returned keys have a defined value.
+ */
+// function getLoginData(account: AccountData) {
+// let loginData: Partial<AccountData> = {};
+// for (const key of ALLOWED_LOGIN_FIELDS) {
+//   loginData[key] = account[key];
+// }
+//   // TODO: account for multiservice when we combine reliers
+//   // const isMultiService = this.relier && this.relier.get('multiService');
+//   // if (isMultiService) {
+//   //   loginData = this._formatForMultiServiceBrowser(loginData);
+//   // }
+//   loginData.verified = !!loginData.verified;
```

Comment:
Q: What does this do?
Are we expecting `verified` to be non-Boolean, hence the re-casting?
Because if it's a string-like `'false'` from the query string, I think the cast will convert to `'false'` to `true`.

```js
window.location.search; // '?bool=false'
let qsp = new URLSearchParams(window.location.search);
console.log(qsp.get('bool')) // 'false'
console.log(!!qsp.get('bool')) // true
```

Reply:
I'm not even sure, this is copied over from content-server🥴 I'll take a closer look in that web channel follow up ticket.